### PR TITLE
Update doc: Rename BackupConfigurationTemplate to BackupBlueprint

### PR DESCRIPTION
**Pull request by hossainemruz**
Tasks:
- [x] Rename `directories` to `path`
- [x] Rename `BackupConfigurationTemplate` to `BackupBlueprint`.
- [x] Explain the backup strategy for Deployment,ReplicaSet,ReplicationController(only single pod. uses leader election), StatefulSet(all pods), DaemonSet(all pod).

### CODE REVIEW

File: docs/guides/latest/auto-backup/workload.md [hunk 1]
```diff
@@ -548,14 +548,14 @@ metadata:
...
-    stash.appscode.com/target-directories: "/etc/config"
+    stash.appscode.com/backup-blueprint: workload-backup-blueprint
+    stash.appscode.com/target-paths: "/etc/config"
     stash.appscode.com/volume-mounts: "dmn-config:/etc/config"
 spec:
   selector:
     matchLabels:
       app: stash-demo
-  template:
+  blueprint:
```
Comment (tamalsaha) [inline docs/guides/latest/auto-backup/workload.md]: I think this field should be `template:`

File: docs/concepts/crds/backupsession.md [hunk 2]
```diff
@@ -96,44 +97,74 @@ A `BackupSession` object has the following fields in the `spec` section:
...
+- **StatefulSet:** Every replica of a StatefulSet mounts different volumes. So, taking a backup from each replica is necessary. In this case, sidecar inside each replica takes backup. Stash identifies **pod-0** as **host-0**, **pod-1** as **host-1**, **pod-2** as **host-2** and so on. Hence, the total number of hosts for a StatefulSet is the number of replicas.
+- **DaemonSet:** Daemon replicas on every node may contain different data. So, taking a backup of each daemon pod is necessary. In this case, sidecar inside each daemon pod takes backup. Stash considers the individual daemon pod as a separate host and the **node name** where the daemon pod is running act as their **host** identifier. The total number of hosts for a DaemonSet is the number of daemon pod running in the cluster.
```
Comment (tamalsaha) [inline docs/concepts/crds/backupsession.md]: DeploymentConfig

File: docs/concepts/crds/backupsession.md [hunk 3]
```diff
@@ -96,44 +97,74 @@ A `BackupSession` object has the following fields in the `spec` section:
...
+    - **unmodifiedFiles:** `unmodifiedFiles` shows the number of files that haven't been changed since last backup of this path.
+- **error:** `error` shows the reason for failure if the backup process failed for this host.
+
+### Hosts of a backup process
+
+Stash uses two different models for backup depending on the target type. It uses **sidecar model** for Kubernetes workloads and **job model** for rest of the targets. In the sidecar model, Stash injects a sidecar inside the targeted workload and the sidecar is responsible for taking backup. In the job model, Stash launches a job to take a backup of the target.
+
+Stash uses an identifier called **host** to separate the backed up data of different subject in the backed. This host identification process depends on the backup model and the target types. The backup strategy and host identification strategy for different types of  target is explained below.
+
+**Kubernetes Workloads:**
```
Comment (tamalsaha) [inline docs/concepts/crds/backupsession.md]: Kubernetes & OpenShift Workloads

File: docs/concepts/crds/appbinding.md [hunk 4]
```diff
@@ -71,7 +72,20 @@ An `AppBinding` object has the following fields in the `spec` section that are r
...
+Here, the variables are parsed as follows:
+
+|       Variable        |                                                   Usage                                                   |
+| --------------------- | --------------------------------------------------------------------------------------------------------- |
+| `TARGET_APP_GROUP`    | Represents the application group where the respective app belongs (i.e: `kubedb.com`).                               |
+| `TARGET_APP_RESOURCE` | Represents the resource kind under that application group that the respective app works with (i.e: `postgres`).       |
+| `TARGET_APP_TYPE`     | Represents the total types of the application. It's simply `TARGET_APP_GROUP/TARGET_APP_RESOURCE` (i.e: `kubedb.com/postgres`). |
+
+
+> The type field
```
Comment (tamalsaha) [inline docs/concepts/crds/appbinding.md]: incomplete line?

File: docs/concepts/crds/restoresession.md [hunk 5]
```diff
@@ -133,7 +126,7 @@ A `RestoreSession` object has the following fields in the `spec` section.
...
+- **spec.target.volumeClaimTemplates :** `spec.target.volumeClaimTemplates` specifies a list of PVC templates that will be created by restoring data from respective VolumeSnapshots. You have to set `spec.dataSource` section to the respective VolumeSnapshot. You can templatize `spec.dataSource.name` section. Stash will resolve the template and dynamically creates respective PVCs and initialize them from respective VolumeSnapshots. Use this field only if `spec.driver` is set to `VolumeSnapshotter`. For more details on how to restore PVCs from VolumeSnapshot, please visit [here](/docs/guides/latest/volumesnapshot/pvc.md).
```
Comment (tamalsaha) [inline docs/concepts/crds/restoresession.md]: I thought we supported this with Restic driver, too.

Comment (hossainemruz) [reply docs/concepts/crds/restoresession.md]: fixed

**Pull request merged**